src/main.rs

```diff
@@ -734,15 +762,40 @@ async fn process_signed_delegate_action(
     if USE_PAY_WITH_FT.clone() {
         let non_delegate_actions = signed_delegate_action.delegate_action.get_actions();
         let treasury_payments: Vec<Action> = non_delegate_actions
-            .into_iter()
-            .filter(|x| matches!(
-                x,
-                Action::FunctionCall(FunctionCallAction { args, .. }
-                ) if String::from_utf8_lossy(args).contains(&BURN_ADDRESS.to_string()))
-            )
+            .iter()
+            .filter_map(|action| {
+                if let Action::FunctionCall(FunctionCallAction { args, .. }) = action {
+                    debug!("args: {:?}", args);
+
+                    // convert to ascii lowercase
+                    let args_ascii = args.to_ascii_lowercase();
+                    debug!("args_ascii: {:?}", args_ascii);
+
+                    // Convert to UTF-8 string
+                    let args_str = String::from_utf8_lossy(&args_ascii);
+                    debug!("args_str: {:?}", args_str);
+
+                    // Parse to JSON (assuming args are serialized as JSON)
+                    let args_json: Value =
+                        serde_json::from_str(&args_str).expect("Failed to parse JSON");
```

Comment:
if this manual way of deserializing can't be avoid, at least let's avoid doing these `expect`s, since it's a potential way to kill the service